autoload/lsp_settings.vim

```diff
@@ -472,6 +472,11 @@ function! s:vim_lsp_load_or_suggest(ft) abort
   if get(g:, 'lsp_loaded', 0)
     for l:server in s:settings[a:ft]
       let l:config = lsp_settings#server_config(l:server.command)
+      if has_key(g:, 'lsp_settings') &&
+      \  has_key(g:lsp_settings, l:server.command) &&
+      \  has_key(g:lsp_settings[l:server.command], 'config')
+        call extend(l:config, g:lsp_settings[l:server.command]['config'])
+      endif
```

Comment:
extend is destructive API. please override `refresh_pattern` simply. (not to use extend)